### PR TITLE
ReadCertificate[Bundle|Request] PEM check whole file for PEM header.

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: minica/minica_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func (p badSigner) Public() crypto.PublicKey {
 	return []byte("foo")
 }
 
-func (p badSigner) Sign(r io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
+func (p badSigner) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
```
Comment (maraino) [inline minica/minica_test.go:26]: Is the linter not failing with digest and opts?

File: sshutil/certificate_test.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func (b *badSigner) Sign(r io.Reader, data []byte) (*ssh.Signature, error) {
 	return nil, fmt.Errorf("an error")
 }
 
-func (b *badSigner) SignWithAlgorithm(r io.Reader, data []byte, algorithm string) (*ssh.Signature, error) {
+func (b *badSigner) SignWithAlgorithm(_ io.Reader, data []byte, algorithm string) (*ssh.Signature, error) {
```
Comment (maraino) [inline sshutil/certificate_test.go:29]: linter didn't catch this?

**Review by maraino — APPROVED** [review]
LGTM.

I disabled auto-merge because I was surprised because the linter missed the unused parameters in `minica_test.go`, `sshutil/certificate_test.go` and perhaps others.

There's also no need to use `_` if all the parameters are unused. For example, the first looks better than the second:

```go
func (k *badSSHPublicKey) Verify([]byte, *ssh.Signature) error
func (k *badSSHPublicKey) Verify(_ []byte, _ *ssh.Signature) error
```

File: sshutil/certificate_test.go (hunk 2 above)
Comment (dopey) [reply sshutil/certificate_test.go:29]: https://github.com/smallstep/workflows/pull/31/files -- stopped using this linter for tests midway through generating the PR.

File: minica/minica_test.go (hunk 1 above)
Comment (dopey) [reply minica/minica_test.go:26]: https://github.com/smallstep/workflows/pull/31/files -- stopped using this linter for tests midway through generating the PR.

**Pull request merged**